### PR TITLE
Notify claim approved

### LINKED ISSUE #238: Send mail to claimant when claim is approved

### PR DESCRIPTION
**Pull request by jdungan**
Should fix #238 and #239.

### CODE REVIEW

File: auctions/models.py [hunk 1]
```diff
@@ -175,12 +175,39 @@ class Vote(models.Model):
...
     def __unicode__(self):
         return u'Vote for %s by (%s): %s' % (
             self.claim, self.user, self.approved
         )
 
 
+@receiver(post_save, sender=Vote)
+def notify_approved_claim(sender, instance, created, **kwargs):
+    claim = instance.claim
+    votes_needed = Bid.objects.filter(issue=claim.issue).filter(offer__gt=0)
```
Comment (groovecoder) [inline auctions/models.py]: Don't we also need to `.exclude(user=claim.user)` ?

File: auctions/tests/model_tests.py [hunk 2]
```diff
@@ -258,6 +259,21 @@ def test_dont_send_email_on_saving_claim(self, mock_send_mail):
...
+                                email='user2@test.com')
+        self.user3 = mommy.make(settings.AUTH_USER_MODEL,
+                                email='user3@test.com')
+        url = 'http://test.com/bug/123'
+        issue = mommy.make(Issue, url=url)
+
+        mommy.make(Bid, user=self.user1, url=url, issue=issue, ask=50)
+        mommy.make(Bid, user=self.user2, url=url, issue=issue, offer=50)
+        mommy.make(Bid, user=self.user3, url=url, issue=issue, offer=50)
+        self.claim = mommy.make(Claim, issue=issue, user=self.user1)
```
Comment (groovecoder) [inline auctions/tests/model_tests.py]: Is all this `setUp` close enough to the `MarketTestCase.setUp` that we could just make `VoteTest` extend from `MarketTestCase` to reduce duplication?

File: auctions/tests/model_tests.py [hunk 3]
```diff
@@ -273,3 +289,14 @@ def test_save_updates_datetimes(self):
                         "Vote.modified should be auto-populated on update.")
         self.assertTrue(datetime.now() >= test_vote_modified,
                         "Vote.modified should be auto-populated.")
+
+    def test_user_claim_unique(self):
+        mommy.make(Vote, claim=self.claim, user=self.user1, approved=True)
+        with self.assertRaises(IntegrityError):
```
Comment (groovecoder) [inline auctions/tests/model_tests.py]: ooo, nice use of `with` ... I should change some of my `try ... catch IntegrityError` type assertions with this.

File: auctions/tests/model_tests.py [hunk 4]
```diff
@@ -273,3 +289,14 @@ def test_save_updates_datetimes(self):
...
+            mommy.make(Vote, claim=self.claim, user=self.user1, approved=False)
+
+    @fudge.patch('auctions.models.send_mail')
+    def test_notify_claim_approved(self, mock_send_mail):
+        mock_send_mail.expects_call()
```
Comment (groovecoder) [inline auctions/tests/model_tests.py]: Should fill in some more scenario tests here, or at least add a `# TODO: Add more test scenarios to VoteTest`

File: auctions/models.py (hunk 1 above)
Comment (jdungan) [inline auctions/models.py]: Yes we do. We should also keep a user from voting on their own claim

File: auctions/tests/model_tests.py (hunk 3 above)
Comment (jdungan) [inline auctions/tests/model_tests.py]: +1 goes to stackoverflow

File: auctions/tests/model_tests.py (hunk 2 above)
Comment (jdungan) [inline auctions/tests/model_tests.py]: We'd need to tweet the MarketTest bids. Notice the explicit issue on the new Bids.

Comment (jdungan) [inline auctions/tests/model_tests.py]: Tweak

**Pull request merged**